airguardian-core/src/main/kotlin/systems/ajax/motrechko/airguardian/controller/grpc/MonitoringObjectGrpcService.kt

```diff
@@ -0,0 +1,135 @@
+
+import net.devh.boot.grpc.server.service.GrpcService
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+import systems.ajax.motrechko.airguardian.ReactorMonitoringObjectServiceGrpc
+import systems.ajax.motrechko.airguardian.commonresponse.monitoring_object.proto.MonitoringObject
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.create_monitoring_object.CreateMonitoringObjectRequest
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.create_monitoring_object.CreateMonitoringObjectResponse
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.delete_monitoring_object.DeleteMonitoringObjectRequest
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.delete_monitoring_object.DeleteMonitoringObjectResponse
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.get_monitoring_object.GetMonitoringObjectRequest
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.get_monitoring_object.GetMonitoringObjectResponse
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.get_monitoring_objects.GetMonitoringObjectsRequest
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.get_monitoring_objects.GetMonitoringObjectsResponse
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.update_monitoring_object.UpdateMonitoringObjectRequest
+import systems.ajax.motrechko.airguardian.input.reqrepl.monitoringobject.update_monitoring_object.UpdateMonitoringObjectResponse
+import systems.ajax.motrechko.airguardian.mapper.toEntity
+import systems.ajax.motrechko.airguardian.mapper.toProto
+import systems.ajax.motrechko.airguardian.service.MonitoringObjectService
+
+@GrpcService
+@Suppress("TooManyFunctions")
+class MonitoringObjectGrpcService(
+    private val monitoringObjectService: MonitoringObjectService
+) : ReactorMonitoringObjectServiceGrpc.MonitoringObjectServiceImplBase() {
+    override fun getMonitoringObject(request: Mono<GetMonitoringObjectRequest>): Mono<GetMonitoringObjectResponse> =
+        request.flatMap { processFindById(it) }
+
+    override fun getMonitoringObjects(request: Mono<GetMonitoringObjectsRequest>): Mono<GetMonitoringObjectsResponse> =
+        request.flatMap { processGetAll() }
+
+    override fun createMonitoringObject(request: Mono<CreateMonitoringObjectRequest>)
+    : Mono<CreateMonitoringObjectResponse> =
+        request.flatMap { processCreate(it) }
+    override fun updateMonitoringObject(request: Mono<UpdateMonitoringObjectRequest>)
+    : Mono<UpdateMonitoringObjectResponse> =
+        request.flatMap { processUpdate(it) }
+
+    override fun deleteMonitoringObject(request: Mono<DeleteMonitoringObjectRequest>)
+            : Mono<DeleteMonitoringObjectResponse> =
+        request.flatMap { processDelete(it) }
+
+    private fun processCreate(createMonitoringObjectRequest: CreateMonitoringObjectRequest) =
```

Comment:
you can extract this method to some CreateMonitoringObjectService/UseCase. With such an approach, you'll have a few small classes, each is responsible for some action. and your MonitoringObjectGrpcService will only delegate execution to them